### PR TITLE
Add examples for reading/writing checkpoints with lightning in various ways

### PR DESCRIPTION
**Pull request by muddyfish**
## Description

Adds four example files for using S3 connector for Pytorch with Pytorch Lightning:

- Reading from a checkpoint
- Writing to checkpoints automatically, including options to keep the most recent checkpoint only.
- Manually writing to a checkpoint
- Asynchronous checkpoint writing

## Additional context

No breaking changes

## Testing

Ran all files, and saw they wrote files, and read files from S3.

--------
By submitting this pull request, I confirm that my contribution is made under the terms of BSD 3-Clause License and I agree to the terms of the [LICENSE](https://github.com/awslabs/s3-connector-for-pytorch/blob/main/LICENSE).

### CODE REVIEW

**Review by IsaevIlya — COMMENTED** [review]
Could we please add the small section about using S3LightningCheckpoint into Readme.md. That we are providing support for Lightning checkpoints, how to install that module, simple example of usage like:
```
s3_checkpoint_io = S3LightningCheckpoint("us-east-1")
trainer = Trainer(
    plugins=[s3_checkpoint_io],
    default_root_dir="s3://bucket_name/key_prefix/"
)
trainer.fit(model)
```

File: examples/lightning/async_checkpoint_writing.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+def main(region: str, checkpoint_path: str):
+    dataset = WikiText2()
+    dataloader = DataLoader(dataset, num_workers=3)
+
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
+    s3_lightning_checkpoint = S3LightningCheckpoint(region)
+    async_checkpoint = AsyncCheckpointIO(s3_lightning_checkpoint)
+
+    checkpoint_callback = ModelCheckpoint(
```
Comment (dnanuti) [inline examples/lightning/async_checkpoint_writing.py:21]: Can we perhaps add some explanatory comments such as:
- checkpoint path will look like ... 
- we expect N checkpoints to be saved ... etc.

Same for the other examples.

File: examples/lightning/checkpoint_reading.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
+    s3_lightning_checkpoint = S3LightningCheckpoint(region)
+
+    trainer = Trainer(
+        plugins=[s3_lightning_checkpoint],
+        min_epochs=4,
+        max_epochs=5,
+        max_steps=3,
+    )
+    # Load the checkpoint in `ckpt_path` before training
```
Comment (dnanuti) [inline examples/lightning/checkpoint_reading.py:24]: nit: load in CHECKPOINT_PATH

**Review by dnanuti — APPROVED** [review]

File: examples/lightning/checkpoint_reading.py (hunk 2 above)
Comment (muddyfish) [reply examples/lightning/checkpoint_reading.py:24]: I was thinking it makes more sense to use `ckpt_path` as that's the name of the argument as supplied by Lightning.

**Pull request merged**